plugins/toolbarwatcher/plugin.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+(function () {
+	'use strict';
+
+	var PLUGIN_NAME = 'toolbarwatcher';
+
+	var stylesLoaded = false;
+
+	if (!CKEDITOR.plugins.get(PLUGIN_NAME)) {
+		var TOOLBAR_HIDDEN_LABELS_CLASS = 'cke_hide_toolbar_labels';
+
+		function debounce(fn, delay) {
+			return function debounced() {
+				var args = Array.prototype.slice.call(arguments);
+				clearTimeout(debounced.id);
+				debounced.id = setTimeout(function () {
+					fn.apply(null, args);
+				}, delay);
+			};
+		}
+
+		CKEDITOR.plugins.add(PLUGIN_NAME, {
+			afterInit: function (editor) {
+				this.editor = editor;
+				this.editor.on(
+					'instanceReady',
+					this.onInstanceReady.bind(this)
+				);
+			},
+
+			editor: null,
+
+			getToolbarsWidth: function () {
+				var containerElement = this.editor.container.$;
+
+				var width = 0;
+
+				containerElement
+					.querySelectorAll('.cke_toolbar')
+					.forEach(function (element) {
+						width += element.offsetWidth;
+					});
+
+				return width;
+			},
+
+			init: function(editor) {
+				if (!stylesLoaded) {
+					CKEDITOR.document.appendStyleSheet(
+						this.path + 'skins/default.css'
+					);
```

Comment:
I thought [you said](https://github.com/liferay/liferay-ckeditor/pull/102#discussion_r465661178):

> Let's embrace `CKEDITOR.addCss`

Reply:
I did, the problem is that sometimes these classes are added to the `iframe` and aren't applied, at least that's what @carloslancha told me

Reply:
I tested this pr before that change and css was not being loaded, online inside the content iframe (where you actually write). Looks like extracting this to a css file did the trick, don't know if @julien found out why 😅

Reply:
@carloslancha the difference is that with `CKEDITOR.addCss` as you said the styles are only applied to CKEDITOR's iframe, with `CKEDITOR.document.appendStyleSheet` they're in the "top" document, i.e.

![image](https://user-images.githubusercontent.com/5572/90009431-58bdd200-dc9e-11ea-9deb-22ef0d42e02c.png)

In blogs it doesn't seem to matter but I wanted to confirm that if we don't do this in wiki or knowledge base it breaks